Gem/Code/Source/MultiplayerSampleAWSGameLiftServerSystemComponent.cpp

```diff
@@ -72,7 +76,36 @@ namespace MultiplayerSample
     bool MultiplayerSampleAWSGameLiftServerSystemComponent::OnCreateSessionBegin([[maybe_unused]] const Multiplayer::SessionConfig& sessionConfig)
     {
         // Add here: additional logic, if needed, to ready server process for hosting a session.
-        // For now, sufficient to return true so Amazon GameLift knows server process is responsive.
+
+        AzFramework::LevelLoadBlockerBus::Handler::BusDisconnect();
+
+        if (m_loadedLevelName)
+        {
+            AZ_Info("MultiplayerSampleAWSGameLiftServerSystemComponent", "Attempting to load level: '%s'", m_loadedLevelName);
+
+            auto loadLevelCommand = AZStd::string::format("LoadLevel %s", m_loadedLevelName);
+            AZ::Interface<AZ::IConsole>::Get()->PerformCommand(loadLevelCommand.c_str());
+        }
+        else
+        {
+            AZ_Info("MultiplayerSampleAWSGameLiftServerSystemComponent", "No level was previously loaded, skipping call to LoadLevel.");
+        }
+
+        return true;
+    }
+
+    bool MultiplayerSampleAWSGameLiftServerSystemComponent::ShouldBlockLevelLoading(const char* levelName)
+    {
+        m_loadedLevelName = levelName;
```

Comment:
Just verifying, is the lifetime of the string pointed to by "levelName" guaranteed to last all the way to whenever the level load happens? i.e. should this string get copied instead of just copying the pointer?

Reply:
Ah great catch! Definitely not, this is passed in from a method’s local string which goes bye-bye.

The question alone I think is enough reason to save a copy to azstd::string. Maybe even change the API to provide a const string&? Seems like an easy mistake to make

Reply:
> is the lifetime of the string pointed to by "levelName" guaranteed to last all the way to whenever the level load happens?

ummm nope! Awesome catch, will change the member variable to a string.